raster-foundry/raster-foundry
# Allow separating annotations into groups

## Overview
Add the ability to create annotation groups and filter by them

### Checklist

- ~Styleguide updated, if necessary~
- ~Any content changes are properly templated using `BUILDCONFIG.APP_NAME`~

### Notes
Still need to add a frontend. This PR should work with the existing frontend and not break stuff

I did not remove projectId from annotations, since I wanted to minimize what I changed for now given that we'll be making a separate annotation application

## Testing Instructions

 * Verify that any projects that have a annotations on them have a group created automatically
* Verify that all the old endpoints still function 
* Create a new annotation group
* Verify that the annotation group list endpoint works
* Add annotations to two separate annotation groups. The frontend doesn't support this yet, so you'll have to use something like postman
* Verify that filtering by annotation groups using the old `/project/{}/annotations` endpoint works (param is `annotationGroup={}`
* Verify that you can create annotations directly in an annotation group, but if one isn't specified, it will place them into a default one defined for the project. If a default group is not defined, it will create a new annotation group called "Annotations" and add all subsequent annotations to it.

Connects https://github.com/raster-foundry/raster-foundry/issues/2849
Not closing the issue, since we don't actually add the frontend at this point.
